
This patch works upon PostgreSQL git tags REL_13_0 through REL_13_2.
However, instead, use the R script: msvc.diff.R
and its call: 
"%R_HOME%\bin\%rbin%\Rscript.exe" --vanilla %APPVEYOR_BUILD_FOLDER%\msvc.diff.R

diff --git a/src/tools/msvc/Mkvcbuild.pm b/src/tools/msvc/Mkvcbuild.pm
index 20da7985c1..d2b95cd71d 100644
--- a/src/tools/msvc/Mkvcbuild.pm
+++ b/src/tools/msvc/Mkvcbuild.pm
@@ -50,7 +50,8 @@ my @contrib_excludes = (
 	'pgcrypto',         'sepgsql',
 	'brin',             'test_extensions',
 	'test_misc',        'test_pg_dump',
-	'snapshot_too_old', 'unsafe_tests');
+	'snapshot_too_old', 'unsafe_tests',
+	'plr');
 
 # Set of variables for frontend modules
 my $frontend_defines = { 'initdb' => 'FRONTEND' };
@@ -474,6 +475,16 @@ sub mkvcbuild
 	my $mf = Project::read_file('contrib/pgcrypto/Makefile');
 	GenerateContribSqlFiles('pgcrypto', $mf);
 
+	my $plr = $solution->AddProject('plr','dll','plr','contrib/plr');
+	$plr->AddFiles(
+		'contrib\plr','plr.c','pg_conversion.c','pg_backend_support.c','pg_userfuncs.c','pg_rsupport.c'
+	);
+	$plr->AddReference($postgres);
+	$plr->AddLibrary('contrib/plr/R$(PlatformTarget).lib');
+	$plr->AddIncludeDir('$(R_HOME)\include');
+	my $mfplr = Project::read_file('contrib/plr/Makefile');
+	GenerateContribSqlFiles('plr', $mfplr);
+
 	foreach my $subdir ('contrib', 'src/test/modules')
 	{
 		opendir($D, $subdir) || croak "Could not opendir on $subdir!\n";
@@ -1030,6 +1041,15 @@ sub GenerateContribSqlFiles
 			}
 		}
 	}
+	else
+	{
+		print "GenerateContribSqlFiles skipping $n\n";
+		if ($n eq 'plr')
+		{
+			print "mf: $mf\n";
+		}
+	}
+
 	return;
 }
 
diff --git a/src/tools/msvc/vcregress.pl b/src/tools/msvc/vcregress.pl
index 4a53a004b7..7a22d4e3ca 100644
--- a/src/tools/msvc/vcregress.pl
+++ b/src/tools/msvc/vcregress.pl
@@ -453,7 +453,7 @@ sub contribcheck
 {
 	chdir "../../../contrib";
 	my $mstat = 0;
-	foreach my $module (glob("*"))
+	foreach my $module (glob("plr"))
 	{
 		# these configuration-based exclusions must match Install.pm
 		next if ($module eq "uuid-ossp"  && !defined($config->{uuid}));
